### PR TITLE
solution

### PR DESCRIPTION
**Pull request by kostya-konev**
(https://kostya-konev.github.io/react_add-todo-form/)

### CODE REVIEW

**Review by alexandra-protyanova — CHANGES_REQUESTED** [review]
Todo shouldn't be added if is empty and error message should be displayed

File: src/App.tsx [hunk 1]
```diff
@@ -1,17 +1,83 @@
...
+import todos from './api/todos';
+
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
 
 const App: React.FC = () => {
+  const [newTask, setNewTask] = useState('');
+  const [userFromForm, setUserFromForm] = useState('Leanne Graham');
```
Comment (SanyaBratashchuk) [inline src/App.tsx:16]: hold userId here

Comment (SanyaBratashchuk) [inline src/App.tsx:16]:
```suggestion
  const [selectedUser, setSelectedUser] = useState('Leanne Graham');
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,17 +1,83 @@
...
 import users from './api/users';
+import todos from './api/todos';
+
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
 
 const App: React.FC = () => {
+  const [newTask, setNewTask] = useState('');
```
Comment (SanyaBratashchuk) [inline src/App.tsx:15]:
```suggestion
  const [title, setTitle] = useState('');
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,17 +1,83 @@
...
+          const todoToAdd = {
+            userId: neededUser.id,
+            id: allTodos[allTodos.length - 1].id + 1,
+            title: newTask,
+            completed: false,
+            user: neededUser,
+          };
+
+          if (todoToAdd.title) {
+            setAllTodos(prev => [...prev, todoToAdd]);
```
Comment (SanyaBratashchuk) [inline src/App.tsx:41]: move it to the separate function

File: src/App.tsx [hunk 4]
```diff
@@ -1,17 +1,83 @@
...
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
 
 const App: React.FC = () => {
+  const [newTask, setNewTask] = useState('');
+  const [userFromForm, setUserFromForm] = useState('Leanne Graham');
+  const [allTodos, setAllTodos] = useState(preparedTodos);
+  const [isEmpty, setIsEmpty] = useState(false);
```
Comment (SanyaBratashchuk) [inline src/App.tsx:18]: You don't need this state, you already have state with title so it can be just a variable, `const isEmpty = title.length === 0`

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**